types/bluebird/index.d.ts

```diff
@@ -490,11 +490,11 @@ declare class Bluebird<R> implements Bluebird.Thenable<R>, Bluebird.Inspection<R
    *
    * *The original object is not modified.*
    */
-  // TODO verify this is correct
+  // TODO disallow non-objects
   // trusted promise for object
-  static props(object: Bluebird<Object>): Bluebird<Object>;
+  static props<T>(object: Bluebird.Thenable<{ [K in keyof T]: Bluebird.Thenable<T[K]> | T[K] }>): Bluebird<{ [K in keyof T]: T[K] }>;
```

Comment:
If we do that it would be the first type alias ever in the entire bluebird.d.ts file.
I believe `ThenableProps` would be a good idea, but if we add that we should add type aliases for all the other redundant code as well.